### PR TITLE
wow: Use textures for bone data

### PR DESCRIPTION
**Pull request by wgreenberg**
This reduces the size of our uniform buffers by a ton

### CODE REVIEW

File: src/WorldOfWarcraft/program.ts [hunk 1]
```diff
@@ -993,15 +1000,33 @@ void calcBillboardMat(inout mat4 m) {
     m[2] = vec4(upVec, 0.0);
 }
 
+mat4 getBoneTransform(uint boneIndex) {
+    vec4 mx = texelFetch(u_TextureBoneMatrix, ivec2(1, boneIndex), 0);
```
Comment (magcius) [inline src/WorldOfWarcraft/program.ts:1004]: You need a `TEXTURE()` here to make it work on WebGPU:

```suggestion
    vec4 mx = texelFetch(TEXTURE(u_TextureBoneMatrix), ivec2(1, boneIndex), 0);
```

File: src/WorldOfWarcraft/render.ts [hunk 2]
```diff
@@ -145,6 +147,7 @@ export class ModelRenderer {
             vertexBufferDescriptors,
             indexBufferFormat,
         });
+        this.boneTexture = new BoneTexture(device, this.model.boneData.length);
```
Comment (magcius) [inline src/WorldOfWarcraft/render.ts:150]: I think you're missing something to destroy this boneTexture? Might want to run this with the leak checker and make sure it's looking good.

**Review by magcius — COMMENTED** [review]
Nice! Thanks for doing this for me.

File: src/WorldOfWarcraft/render.ts (hunk 2 above)
Comment (wgreenberg) [reply src/WorldOfWarcraft/render.ts:150]: fixed!

File: src/WorldOfWarcraft/program.ts [hunk 3]
```diff
@@ -993,15 +1000,33 @@ void calcBillboardMat(inout mat4 m) {
     m[2] = vec4(upVec, 0.0);
 }
 
+mat4 getBoneTransform(uint boneIndex) {
+    vec4 mx = texelFetch(TEXTURE(u_TextureBoneMatrix), ivec2(1, boneIndex), 0);
```
Comment (magcius) [inline src/WorldOfWarcraft/program.ts:1004]: Sorry, the TEXTURE() comment should apply to all the texelFetch calls in here.

Comment (wgreenberg) [reply src/WorldOfWarcraft/program.ts:1004]: oh that makes sense lol

**Pull request merged**